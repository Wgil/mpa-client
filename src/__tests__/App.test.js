import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

jest.mock('../components/About');
jest.mock('../components/Archive');
jest.mock('../components/Home');
jest.mock('../components/Login');
jest.mock('../components/Signup');
jest.mock('../components/ResetPassword');
jest.mock('../components/FourOhFour');

import App from '../AppRouter';
import About from '../components/About';
About.mockReturnValue(() => <div />);
import Archive from '../components/Archive';
Archive.mockReturnValue(() => <div />);
import Home from '../components/Home';
Home.mockReturnValue(() => <div />);
import Login from '../components/Login';
Login.mockReturnValue(() => <div />);
import Signup from '../components/Signup';
Signup.mockReturnValue(() => <div />);
import ResetPassword from '../components/ResetPassword';
ResetPassword.mockReturnValue(() => <div />);
import FourOhFour from '../components/FourOhFour';
FourOhFour.mockReturnValue(() => <div />);

describe('Test AppRouter.js', () => {
  it('Component Rendered', () => {
    const wrapper = shallow(<App />);
    expect(wrapper).toBeTruthy();
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
