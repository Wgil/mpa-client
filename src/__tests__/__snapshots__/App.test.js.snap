// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test AppRouter.js Component Rendered 1`] = `
<BrowserRouter>
  <section>
    <Switch>
      <Route
        component={[Function]}
        exact={true}
        path="/"
      />
      <Route
        component={[Function]}
        path="/about"
      />
      <Route
        component={[Function]}
        path="/archive"
      />
      <Route
        component={[Function]}
        path="/login"
      />
      <Route
        component={[Function]}
        path="/signup"
      />
      <Route
        component={[Function]}
        path="/resetpass"
      />
      <Route
        component={[Function]}
      />
    </Switch>
  </section>
</BrowserRouter>
`;
